plugins/module_utils/api.py

```diff
@@ -262,6 +264,25 @@ def create(self, resource, data=None):
             resource['state'] = "present"
         return resource
 
+    def wait_for_state(self, check_parameter, allowed_states):
+        start = datetime.now()
+        timeout = self._module.params['api_timeout'] * 2
+        while datetime.now() - start < timedelta(seconds=timeout):
+            info = self.query()
+            if info.get(check_parameter) in allowed_states:
+                return info
+            sleep(1)
+
+        # Timeout succeeded
+        if info.get('name') is not None:
+            msg = "Timeout while waiting for a state change on load balancer %s to states %s. " \
```

Comment:
As this is in the api module it should not mention "load balancer":
```suggestion
            msg = "Timeout while waiting for a state change on resource %s to states %s. " \
```